internal/cmd/test.go

```diff
@@ -135,3 +137,43 @@ func runCMD(mainFunc func(), args []string) (err error) {
 
 	return err
 }
+
+type CommonCMDTestCases struct {
+	ExpectedExitCode int
+	ExpectedStdout   string
+	ExpectedStderr   string
+}
+
+func RunCommonCMDTests(test *testing.T, mainFunc func(), args []string, commonCases CommonCMDTestCases, prefix string) (string, string, int) {
+	stdout, stderr, exitCode, err := RunCMDTest(test, mainFunc, args)
+	if err != nil {
+		test.Errorf("%sCMD run returned an error: '%v'.", prefix, err)
+	}
+
+	if len(stderr) > 0 && commonCases.ExpectedStderr == "" {
+		test.Errorf("%sCMD has content in stderr: '%s'.", prefix, stderr)
+	}
+
+	if !strings.Contains(stderr, commonCases.ExpectedStderr) {
+		test.Errorf("%sUnexpected stderr. Expected substring: '%s', Actual stderr: '%s'.", prefix, commonCases.ExpectedStderr, stderr)
+	}
```

Comment:
Why not a direct equality check?

I can see some more complex cases that could come up (you see these in the Python test infrastructure),
but I don't see those for peek and list.
(And this wouldn't be the way to handle it.)